libnmstate/nm/profile.py

```diff
@@ -21,14 +21,948 @@
+        else:
+            flags |= NM.SettingsUpdate2Flags.IN_MEMORY
+        action = f"Update profile: {self.profile.get_id()}"
+        user_data = action
+        args = None
+
+        self._ctx.register_async(action, fast=True)
+        self.profile.update2(
+            self.profile.to_dbus(NM.ConnectionSerializationFlags.ALL),
+            flags,
+            args,
+            self._ctx.cancellable,
+            self._update2_callback,
+            user_data,
+        )
+
+    def add(self, save_to_disk=True):
+        nm_add_conn2_flags = NM.SettingsAddConnection2Flags
+        flags = nm_add_conn2_flags.BLOCK_AUTOCONNECT
+        if save_to_disk:
+            flags |= nm_add_conn2_flags.TO_DISK
+        else:
+            flags |= nm_add_conn2_flags.IN_MEMORY
+
+        action = f"Add profile: {self.profile.get_id()}"
+        self._ctx.register_async(action, fast=True)
+
+        user_data = action
+        args = None
+        ignore_out_result = False  # Don't fall back to old AddConnection()
+        self._ctx.client.add_connection2(
+            self.profile.to_dbus(NM.ConnectionSerializationFlags.ALL),
+            flags,
+            args,
+            ignore_out_result,
+            self._ctx.cancellable,
+            self._add_connection2_callback,
+            user_data,
+        )
+
+    def delete(self):
+        if self._profile:
+            action = (
+                f"Delete profile: id:{self.profile.get_id()}, "
+                f"uuid:{self.profile.get_uuid()}"
+            )
+            user_data = action
+            self._ctx.register_async(action, fast=True)
+            self.profile.delete_async(
+                self._ctx.cancellable,
+                self._delete_connection_callback,
+                user_data,
+            )
+
+    def activate(self):
+        if self.uuid:
+            self.import_by_uuid()
+        elif not self.profile:
+            raise NmstateInternalError(
+                "BUG: Failed  to find valid profile to activate: "
+                f"id={self.con_id}, dev={self.devname} uuid={self.uuid}"
+            )
+
+        specific_object = None
+        if self.profile:
+            action = f"Activate profile: {self.profile.get_id()}"
+        elif self.nmdevice:
+            action = f"Activate profile: {self.nmdevice.get_iface()}"
+        else:
+            raise NmstateInternalError(
+                "BUG: Cannot activate a profile with empty profile id and "
+                "empty NM.Device"
+            )
+        user_data = action
+        self._ctx.register_async(action)
+        self._ctx.client.activate_connection_async(
+            self.profile,
+            self.nmdevice,
+            specific_object,
+            self._ctx.cancellable,
+            self._active_connection_callback,
+            user_data,
+        )
+
+    @property
+    def profile(self):
+        return self.profile
+
+    @profile.setter
+    def profile(self, con_profile):
```

Comment:
In stead exposing old interfaces, you should think about expose minimum interfaces required, so that you could know which is old unused function.